js/tests/launchAsync.test.ts

```diff
@@ -271,6 +271,47 @@ describe('launchAsync tests', () => {
 
         expect(cbOnExit).toHaveBeenCalledTimes(1);
     });
+    
+    it('launches with preferences not set', async () => {
+        expect.assertions(1);
+        const options: Options = { preferences: null };
+        const launchPromise = launchAsync(SampleToken, SampleSubdomain, SampleContent, options)
+            .then(iframe => {
+                expect(iframe).not.toBeNull();
+            });
+        window.postMessage('ImmersiveReader-LaunchResponse:{"success":true}', '*');
+
+        return launchPromise;
+    });
+    
+    it('launches with preferences set', async () => {
+        expect.assertions(1);
+        const options: Options = { preferences: 'foo' };
+        const launchPromise = launchAsync(SampleToken, SampleSubdomain, SampleContent, options)
+            .then(iframe => {
+                expect(iframe).not.toBeNull();
+            });
+        window.postMessage('ImmersiveReader-LaunchResponse:{"success":true}', '*');
+
+        return launchPromise;
+    });
+    
+    it('launches with onPreferencesChanged set', async () => {
+        jest.useRealTimers();
+        expect.assertions(1);
+        const options: Options = { onPreferencesChanged: (value) => { 
+            expect(value).toBe('hello world');
```

Comment:
can we add something to verify this actually gets called? Set a bool and check it after the await?

Reply:
the expect.assertions(1) is what verifies it